src/Diagnostics.ModelsAndUtils/Models/ResponseExtensions/Notification.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace Diagnostics.ModelsAndUtils.Models.ResponseExtensions
+{
+    /// <summary>
+    /// Class representing Notification
+    /// </summary>
+    public class Notification
+    {
+        /// <summary>
+        /// Enum representing Notification level.
+        /// </summary>
+        public InsightStatus Status;
+
+        /// <summary>
+        /// Notification title.
+        /// </summary>
+        public string Title;
+
+        /// <summary>
+        /// Notification description.
+        /// </summary>
+        public string Description;
+
+
+        /// <summary>
+        /// Optional solutions to common problems to which the notification may refer
+        /// </summary>
+        public Solution Solution;
+
+
+        /// <summary>
+        /// Whether a notification is expanded to begin with
+        /// </summary>
+        public bool IsExpanded;
+
+        /// <summary>
+        /// The startdate when the notification will be shown
+        /// </summary>
+        public DateTime StartDate;
+
+
+        /// <summary>
+        /// The end date to stop showing the notification
+        /// </summary>
+        public DateTime ExpiryDate;
+
+
+        /// <summary>
+        /// Creates an instance of notification class.
+        /// </summary>
+        /// <param name="status">Enum reprensenting notification level.</param>
+        /// <param name="title">Notification title.</param>
+        /// <param name="description">Notification description.</param>
+        /// <param name="solution">Solution to address the notification.</param>
+        /// <param name="isExpanded">Whether a notification is expanded to begin with.</param>
+        /// <param name="startDate">The startdate when the notification will be shown.</param>
+        /// <param name="expiryDate">The end date to stop showing the notification.</param>
+        public Notification(InsightStatus status, string title, string description = "", Solution solution = null, bool isExpanded = true, DateTime? startDate = null, DateTime? expiryDate = null)
+        {
+            Status = status;
+            Title = title;
+            Description = description;
+            Solution = solution;
+            IsExpanded = isExpanded;
```

Comment:
also, how do you plan to use isExpanded for notification? My understanding is this will show as a notification bar on the top

Reply:
This rendering is not for the message bar Shekhar. It's the rendering for the detailed notification in the side panel:
I am still debating about whether we should have this option or not:

![image](https://user-images.githubusercontent.com/38216903/108462168-03581400-7231-11eb-832f-47142684a878.png)